pkgs/development/python-modules/triton/default.nix

```diff
@@ -139,40 +89,38 @@ buildPythonPackage rec {
   ];
 
   # Avoid GLIBCXX mismatch with other cuda-enabled python packages
-  preConfigure =
-    ''
-      # Ensure that the build process uses the requested number of cores
-      export MAX_JOBS="$NIX_BUILD_CORES"
+  preConfigure = ''
+    # Ensure that the build process uses the requested number of cores
+    export MAX_JOBS="$NIX_BUILD_CORES"
 
-      # Upstream's setup.py tries to write cache somewhere in ~/
-      export HOME=$(mktemp -d)
+    # Upstream's setup.py tries to write cache somewhere in ~/
+    export HOME=$(mktemp -d)
 
-      # Upstream's github actions patch setup.cfg to write base-dir. May be redundant
-      echo "
-      [build_ext]
-      base-dir=$PWD" >> python/setup.cfg
+    # Upstream's github actions patch setup.cfg to write base-dir. May be redundant
+    echo "
+    [build_ext]
+    base-dir=$PWD" >> python/setup.cfg
 
-      # The rest (including buildPhase) is relative to ./python/
-      cd python
-    ''
-    + lib.optionalString cudaSupport ''
-      export CC=${cudaPackages.backendStdenv.cc}/bin/cc;
-      export CXX=${cudaPackages.backendStdenv.cc}/bin/c++;
+    # The rest (including buildPhase) is relative to ./python/
+    cd python
+  '';
 
-      # Work around download_and_copy_ptxas()
-      mkdir -p $PWD/triton/third_party/cuda/bin
-      ln -s ${ptxas} $PWD/triton/third_party/cuda/bin
-    '';
+  env = {
+    TRITON_BUILD_PROTON = "OFF";
+  } // lib.optionalAttrs cudaSupport {
+    CC = "${cudaPackages.backendStdenv.cc}/bin/cc";
+    CXX = "${cudaPackages.backendStdenv.cc}/bin/c++";
+
+    TRITON_PTXAS_PATH = lib.getExe' cudaPackages.cuda_nvcc "ptxas"; # Make sure cudaPackages is the right version each update (See python/setup.py)
+    TRITON_CUOBJDUMP_PATH = cudaPackages.cuda_cuobjdump;
+    TRITON_NVDISASM_PATH = cudaPackages.cuda_nvdisasm;
+    TRITON_CUDACRT_PATH = cudaPackages.cuda_nvcc;
+    TRITON_CUDART_PATH = cudaPackages.cuda_cudart;
+    TRITON_CUPTI_PATH = cudaPackages.cuda_cupti;
```

Comment:
Are these actually used? As is these expand into the near-empty `.out` outputs

Reply:
Thats a thing that I would like to know too.

Reply:
As mentioned in the other config, the way TRITON_OFFLINE_BUILD is currently implemented most of these are ignored